components/omega/test/infra/ErrorTest.cpp

```diff
@@ -0,0 +1,157 @@
+   R8 MsgArgReal         = 1.23456789012;
+   I4 MsgArgInt          = 5;
+   std::string MsgArgStr = "Random";
+   if (!Err.isSuccess())
+      ERROR_CRITICAL("Expected critical error with args: {} {} {}", MsgArgStr,
+                     MsgArgInt, MsgArgReal);
+   // Alternative critical failure tests
+   // ERROR_CHECK_CRITICAL(Err, "Expected critical error with args: {} {} {}",
+   //                     MsgArgStr, MsgArgInt, MsgArgReal);
+   // ERROR_REQUIRE(!Err.isSuccess(),
+   //              "Expected Error with args: {} {} {}", MsgArgStr, MsgArgInt,
+   //               MsgArgReal);
+   // This one requires a debug build
+   // ERROR_ASSERT(!Err.isSuccess(),
+   //             "Expected Error with args: {} {} {}", MsgArgStr, MsgArgInt,
+   //             MsgArgReal);
+}
+
+void passError1(Error &Err) { passError2(Err); }
+
+//------------------------------------------------------------------------------
+// Main test driver
+int main(int argc, char **argv) {
+
+   int RetVal = 0;
+
+   MPI_Init(&argc, &argv);
+
+   MachEnv::init(MPI_COMM_WORLD);
+   MachEnv *DefEnv = OMEGA::MachEnv::getDefault();
+   initLogging(DefEnv);
+
+   LOG_INFO("------ Error handler test ------");
+   // Test default constructor and success check
+   Error ErrDefault;
+   if (ErrDefault.isSuccess()) {
+      LOG_INFO("Default error constructor: PASS");
+   } else {
+      LOG_ERROR("Default error constructor: FAIL");
+   }
+
+   // Test constructors with no message and boolean tests
+   Error ErrorTestSuccess2(ErrorCode::Success);
+   if (ErrorTestSuccess2.isSuccess() and !ErrorTestSuccess2.isFail()) {
+      LOG_INFO("No message error constructor success: PASS");
+   } else {
+      LOG_ERROR("No message error constructor success: FAIL");
+   }
+   Error ErrorTestWarn2(ErrorCode::Warn);
+   if (!ErrorTestWarn2.isSuccess() and !ErrorTestWarn2.isFail() and
+       ErrorTestWarn2.Code == ErrorCode::Warn) {
+      LOG_INFO("No message error constructor warn: PASS");
+   } else {
+      LOG_ERROR("No message error constructor warn: FAIL");
+   }
+   Error ErrorTestFail2(ErrorCode::Fail);
+   if (ErrorTestFail2.isFail() and !ErrorTestFail2.isSuccess()) {
+      LOG_INFO("No message error constructor fail: PASS");
+   } else {
+      LOG_ERROR("No message error constructor fail: FAIL");
+   }
+
+   // Test constructor with message
+   // Create a total error to accumulate codes and messages
+   std::string ExpectMsg =
+       "   [error] [ErrorTest.cpp:121] Initialized to fail String 1\n";
+   Error TotalError(ErrorCode::Fail, __LINE__, __FILE__,
+                    "Initialized to fail {} {}", "String", 1);
+   if (TotalError.isFail() and TotalError.Msg == ExpectMsg) {
+      LOG_INFO("Error constructor with message: PASS");
+   } else {
+      LOG_ERROR("Error constructor with message: FAIL");
+      LOG_ERROR(" Error constructor with message expected: {}", ExpectMsg);
+      LOG_ERROR(" Error constructor with message actual:   {}", TotalError.Msg);
+   }
```

Comment:
The following log file messages indicate that this test has actually failed:
```
[* error] [ErrorTest.cpp:124] Error constructor with message: FAIL
[* error] [ErrorTest.cpp:125]  Error constructor with message expected:    [error] [ErrorTest.cpp:121] Initialized to fail String 1

[* error] [ErrorTest.cpp:126]  Error constructor with message actual:      [error] [ErrorTest.cpp:119] Initialized to fail String 1
```

Reply:
I fixed the proximate cause, but as above, I need to find a better way to trap these.

Reply:
Fixed with latest commit